test/test_storage.cpp

```diff
@@ -1016,17 +1016,18 @@ file_storage make_fs()
 	fs.add_file(combine_path("readwritev", "3"), 81);
 	fs.add_file(combine_path("readwritev", "4"), 6561);
 	fs.set_piece_length(0x1000);
-	fs.set_num_pieces((fs.total_size() + 0xfff) / 0x1000);
+	fs.set_num_pieces((int(fs.total_size()) + 0xfff) / 0x1000);
```

Comment:
I don't think this is right.. you shouldn't narrow this until _after_ the division